File: angular/progression/src/app/models/ModuleTask.ts
import { TaskType, RPAGType } from './enums';
import { Module } from './Module';
import { Progression } from './Progression';

export interface ModuleTask {
    moduleTaskID: number;
    moduleId: number;
    taskType: TaskType;
    rpagType: RPAGType;
    inSumRpag: boolean;
    taskName: string;
    url: string;
    dueWeek: number;
    dueLesson: number;
    dueDate: Date | string | null;

    module: Module;

    Progressions: Progression[];
}

    

File: dotnet/progressive/Models/ModuleTask.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace progressive.Models
{

    public class ModuleTask
    {
        // [DatabaseGenerated(DatabaseGeneratedOption.None)]
        public int ModuleTaskID { get; set; }
        public int ModuleID { get; set; }
        public int Term { get; set; }
        public TaskType TaskType { get; set; }
        public RPAGType RPAGType { get; set; }
        public bool InSumRPAG { get; set; }
        public string TaskName { get; set; }
        public string Url { get; set; }
        public int DueWeek { get; set; }
        public int DueLesson { get; set; }
        public DateTime? DueDate { get; set; }

        public Module Module { get; set; }

        public ICollection<Progression> Progressions { get; set; }
    }
}

